src/classes/AccountModelHealthCheckVMapper.cls

```diff
@@ -0,0 +1,394 @@
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************************/
+    protected AccountModelHealthCheckVMapper() {}
+
+    /**********************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of AccountModelHealthCheckVMapper.
+    **********************************************************************************/
+    public static AccountModelHealthCheckVMapper getInstance() {
+        if (instance == null) {
+            instance = new AccountModelHealthCheckVMapper();
+        }
+
+        return instance;
+    }
+
+    /**********************************************************************************
+    * @description Retrieve the results for a for an AccountModel health check.
+    * @return A HealthCheckGroupVModel representing a HealtCheckDescription for Account.
+    **********************************************************************************/ 
+    public virtual HealthCheckGroupVModel getHealthCheckGroup() {
+        AccountModelSettingsModel acctModSettingMod = this.locateAccountModelSettingsService().getAccountModelSettingsModel();
+        Map<Id, RecordTypeModel> accountRTModelsById = this.locateRecordTypeService().getModelByIdMapForSObjectType(Account.SObjectType);
+
+        List<HealthCheckGroupVModel> accountHlthCheckGroupVModelsList = new List<HealthCheckGroupVModel>();
+
+        accountHlthCheckGroupVModelsList.add(this.getDefaultAccountModelHealthCheckItems(acctModSettingMod, accountRTModelsById));        
+        accountHlthCheckGroupVModelsList.add(this.getAdministrativeAccountModelHealthCheckItems(acctModSettingMod, accountRTModelsById));        
+        accountHlthCheckGroupVModelsList.add(this.getHouseholdAccountModelHealthCheckItems(acctModSettingMod, accountRTModelsById));        
+        accountHlthCheckGroupVModelsList.add(this.getAccountRecordTypeUniquenessHealthCheckItems(acctModSettingMod));        
+
+        return this.combineAccountHealthCheckGroupViewModels(accountHlthCheckGroupVModelsList);
+    }
+
+    /**********************************************************************************
+    * @description Generates a HealthCheckGroupVModel comprised of the combined health 
+    * check results for Default Account, Administrative and Household accounts. 
+    * @param  healthCheckGroupVModelsList A list of HealthCheckGroupVModels
+    * @return A HealthCheckGroupVModel representing the health check results for 
+    * Default, Administrative and Household Accounts.
+    **********************************************************************************/ 
+    @TestVisible
+    private HealthCheckGroupVModel combineAccountHealthCheckGroupViewModels(List<HealthCheckGroupVModel> healthCheckGroupVModelsList) {
```

Comment:
This method sounds like one we can extract and reused by all vmappers? maybe we can have a HealthCheckVMapper and let other vmappers to inherit this method?

Reply:
Let's explore the level of effort on it. I'd like to see this used as it is a cleaner design than the one I applied in the Reciprocal Relationship Health Check. In an ideal world this would live in an abstract class that these classes depend on.